plugins/modules/zos_unarchive.py

```diff
@@ -0,0 +1,1063 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2020, 2022
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: zos_unarchive
+version_added: "1.7.0"
+author:
+  - Oscar Fernando Flores Garcia (@fernandofloresg)
+short_description: Unarchive a dataset or file in z/OS.
+description:
+  - The C(zos_unarchive) module unpacks an archive after optionally sending it to the remote.
+
+options:
+  path:
+    description:
+      - Local or remote absolute path or data set name of the archive to be unpacked on the remote.
+    type: str
+    required: true
+  format:
+    description:
+      - The type of compression to use.
+    type: dict
+    required: true
+    suboptions:
+      name:
+          description:
+            - The name of the format to use.
+          type: str
+          required: true
+          choices:
+            - bz2
+            - gz
+            - tar
+            - zip
+            - terse
+            - xmit
+            - pax
+      format_options:
+          description:
+            - Options specific to each format.
+          type: dict
+          required: false
+          suboptions:
+            xmit_log_dataset:
+              description: Provide a name of data set to store xmit log output.
+              type: str
+            use_adrdssu:
+              description:
+                - If set to true, after unpacking a data set in C(xmit) or C(terse) format
+                  it will perform a single DFSMSdss ADRDSSU RESTORE step.
+              type: bool
+              default: False
+            dest_volumes:
+              description: When using ADRDSSU select on which volume the datasets will be placed first.
+              type: list
+              elements: str
+  dest:
+    description:
+    - Remote absolute path where the archive should be unpacked.
+    - The given path must exist. Base directory is not created by this module.
+    type: str
+    required: false
+  group:
+    description:
+      - Name of the group that should own the filesystem object, as would be fed to chown.
```

Comment:
```
  group:
    description:
      - Name of the group that will own the file system objects.
      - When left unspecified, it uses the current group of the current user unless you are root,
         in which case it can preserve the previous ownership.
      - This option is only applicable if C(dest) is USS, otherwise ignored.
    type: str
    required: false
```